src/components/Header.tsx

```diff
@@ -39,25 +53,51 @@ const Header = (props: BoxProps) => {
                 textDecorationColor: 'primary.900',
               }}
             >
-              <HStack spacing={2}>
+              <HStack spacing={{ base: 1, lg: 2 }}>
                 <Box h={LOGO_SIZES} w={LOGO_SIZES}>
-                  <Image src={mascotIcon} alt="" height={150} width={150} />
+                  <Image
+                    src={mascotIcon}
+                    alt=""
+                    height={70}
+                    width={70}
+                    aria-hidden
+                  />
                 </Box>
                 <Heading
                   as="h1"
                   color="primaryText"
                   letterSpacing="tight"
                   fontWeight="black"
-                  fontSize={{ base: '16px', sm: '36px' }}
+                  fontSize={{ base: '16px', md: '36px' }}
                 >
                   Octoclairvoyant
                 </Heading>
               </HStack>
             </RouteLink>
           </Flex>
-          {!!process.env.NEXT_PUBLIC_FEATURE_FLAG_COLOR_MODE && (
-            <ToggleColorModeButton />
-          )}
+          <HStack spacing={{ base: 4, md: 10 }}>
+            {!!process.env.NEXT_PUBLIC_FEATURE_FLAG_COLOR_MODE && (
+              <ToggleColorModeButton
+                boxSize={LOGO_SIZES}
+                minWidth={LOGO_SIZES}
+              />
+            )}
+            <Link
+              href={REPO_URL}
+              aria-label="Octoclairvoyant repository on GitHub"
+              isExternal
+              rounded="full"
+              boxSize={LOGO_SIZES}
+            >
+              <Icon
+                as={DiGithubBadge}
+                boxSize={LOGO_SIZES}
+                color="monochrome1"
+                rounded="full"
+                _hover={{ bgColor: 'monochromeInverted4' }}
```

Comment:
We can probably copy/paste the _active state_ background from the light/dark toggle. So the background darkens a bit if you click + hold the mouse button over the GitHub icon.

To clarify:

- GitHub icon itself doesn't change when hovering/clicking
- The background color on hover effect darkens a bit when you click/press enter

Reply:
I don't get what you mean, sorry. The theme toggle is changing its bg color on hover. Do we want to apply the very same effect to the GitHub one?

Reply:
Oh, maybe this is clearer?

In steps:

1. On hover over GitHub link:
  - Icon stays the same color
  - Round background is a light gray color
2. When clicking on the GitHub link:
   - Icon stays the same color
   - Round background is the same darker gray color as when you click on the theme toggle
   
Let me know if you want me to mock this up in Figma to clarify further. 😉

Reply:
Got it! I'll apply the change in a bit.

Reply:
It should be fixed now.

Reply:
I like it! 😄